src/main/java/io/jsonwebtoken/BadAudienceJwtException.java

```diff
@@ -0,0 +1,11 @@
+package io.jsonwebtoken;
+
+public class BadAudienceJwtException extends ClaimJwtException {
```

Comment:
I think these should be more generic, instead of an exception per condition.

Perhaps something like this:

```java
public class InvalidClaimException extends ClaimJwtException { ... }

//when a value is expected, but one isn't provided in the JWT:
public class MissingClaimException extends InvalidClaimException { ... }

//when a value is expected and one is provided, but the provided value doesn't match the expected value:
public class IncorrectClaimException extends InvalidClaimException {...}
```
(ClaimJwtException should probably be renamed to ClaimException, but that's another issue)

Thoughts?